doc/specs/fftpack.md

```diff
@@ -1280,53 +1213,63 @@ Transformed data to invert.
 This argument is `intent(in)` and `optional`.  
 Defines the length of the Fourier transform. If `n` is not specified (the default) then `n = size(x)`. If `n <= size(x)`, `x` is truncated, if `n > size(x)`, `x` is zero-padded.
 
+`t`: Shall be an `integer` scalar, equal to `1` or `2`.
+This argument is `intent(in)` and `optional`.
+Defines the type of the IDCT to be performed. The default type is `2`.
+
 #### Return value
 
-Returns a `real` and rank-1 array, the inverse Discrete-Cosine Transform (iDCT) of `x`.
+Returns a `real` and rank-1 array, the IDCT type-`t` of the input data `x`.
 
-#### Example
+#### Notes
+
+Within numerical accuracy,
+- `x == idct(dct(x, t=1), t=1) / (2*(size(x) - 1))`
+- `x == idct(dct(x, t=2), t=2) / (4*size(x))`
+- `x == idct(dct(x, t=3), t=3) / (4*size(x))`
+
+### Example
 
 ```fortran
 program demo_idct
     use fftpack, only: dct, idct
     real(kind=8) :: x(4) = [1, 2, 3, 4]
-    print *, idct(dct(x))/(2*(4-1))   !! (normalized):   [1.0, 2.0, 3.0, 4.0]
-    print *, idct(idct(x))/(2*(4-1))  !! (normalized):   [1.0, 2.0, 3.0, 4.0]
-    print *, idct(dct(x), 3)          !! (unnormalized): [7.0, 15.0, 23.0]
+    print *, idct(dct(x,t=1),t=1)/(2*(4-1))   !! (normalized):   [1.0, 2.0, 3.0, 4.0]
+    print *, idct(dct(x,t=2),t=2)/(4*4)       !! (normalized):   [1.0, 2.0, 3.0, 4.0]
+    print *, idct(dct(x), n=3)                !! (unnormalized): [22.06, 32.5, 65.65]
 end program demo_idct
 ```
 
```

Comment:
Since I'm really not clear about the terminology of `dct`, and others may be, perhaps a Wikipedia reference to DCT could be added here:
```markdown
## References
[1] Wikipedia, "Discrete cosine transform", [https://en.wikipedia.org/wiki/Discrete_cosine_transform](https://en.wikipedia.org/wiki/Discrete_cosine_transform)
```

Reply:
Very good point, I'll add the reference.